test/integration2/util/knex-instance-provider.js

```diff
@@ -118,6 +118,12 @@ const testConfigs = {
       user: 'testuser',
       password: 'testpassword',
       charset: 'utf8',
+      typeCast: function (field, next) {
+        if (field.type == 'JSON') {
+          return JSON.parse(field.string());
+        }
+        return next();
+      },
```

Comment:
The `mysql` client does allow for automatic parsing of json fields, but it's opt-in only via the typeCast argument. (see https://github.com/mysqljs/mysql/issues/1267) This means that _users_ who want this behavior will also have to do it this way. We could conceivably make this the default, but I seem to recall you had a preference against doing that sort of thing.